ctapipe/io/hdf5eventsource.py

```diff
@@ -461,6 +472,19 @@ def _generator(self):
                     for table in self.file_.root.dl1.event.telescope.parameters
                 }
 
+        if self.has_muon_parameters:
```

Comment:
Define the muon readers before as empty dict. Makes the code below easier, as you don't have to check, just loop over the dict and it works fine for the empty dict.

Reply:
I do not quite understand how this is meant